### PR TITLE
web: add FLoC environment variable

### PR DESCRIPTION
**Pull request by elrido**
Dear Jitsi Meet community,

This PR is intended to give users of the Jitsi Web container image an option to Opt-In their site's visitors to a new feature being rolled out in Google Chrome called FLoC (Federated Learning of Cohorts).

FLoC is intended to be a more privacy oriented tracking mechanism then what was previously possible with (now phased out) third-party cookies. IMHO this is still a tracking mechanism, even if it is supposed to be more benign then other tracking methods. Further background on FLoC can be found at the [EFF](https://www.eff.org/deeplinks/2021/03/googles-floc-terrible-idea), [W3C](https://wicg.github.io/floc/) or [Google](https://github.com/google/ads-privacy/blob/master/proposals/FLoC/FLOC-Whitepaper-Google.pdf), to offer some different view points. As far as we know, at this time this feature is only implemented by Google Chrome and not yet enabled for all Chrome users, only for a random subset of Chrome users by way of an experiment.

This PR suggests to introduce a new environment variable `DISABLE_FLOC`. If not set or set to `1`, the following HTTP header is added in the nginx configuration, sent in reply to any request: `Permissions-Policy interest-cohort=()`. The intent of this new flag is to disable this tracking mechanism by default, but provide Jitsi admins a mechanism to Opt-In their site's visitors to this tracking method, by setting `DISABLE_FLOC=0`. This will remove that HTTP header and the Chrome browser may start to include visits to the site into their cohort learning.

I attempted to update all files that refer to environment variables. Please advise if there are any documentations that this should get included into or if you have any policies on the naming or ordering of such variables. And of course any other suggestions for improving this PR, if you consider it for inclusion.

### CODE REVIEW

File: examples/traefik-v2/docker-compose.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ services:
             - ENABLE_HSTS
             - ENABLE_XMPP_WEBSOCKET
             - DISABLE_HTTPS
+            - DISABLE_FLOC
```
Comment (saghul) [inline examples/traefik-v2/docker-compose.yml:17]: Please don't add it here, the traefik files will be moved out

File: examples/traefik/docker-compose.yml [hunk 2]
```diff
@@ -16,6 +16,7 @@ services:
             - ENABLE_HTTP_REDIRECT
             - ENABLE_TRANSCRIPTIONS
             - DISABLE_HTTPS
+            - DISABLE_FLOC
```
Comment (saghul) [inline examples/traefik/docker-compose.yml:19]: Ditto

File: web/Dockerfile [hunk 3]
```diff
@@ -9,8 +9,8 @@ RUN \
 	apt-dpkg-wrap apt-get update && \
 	apt-dpkg-wrap apt-get install -y cron nginx-extras jitsi-meet-web socat && \
 	apt-dpkg-wrap apt-get -d install -y jitsi-meet-web-config && \
-    dpkg -x /var/cache/apt/archives/jitsi-meet-web-config*.deb /tmp/pkg && \
-    mv /tmp/pkg/usr/share/jitsi-meet-web-config/config.js /defaults && \
+	dpkg -x /var/cache/apt/archives/jitsi-meet-web-config*.deb /tmp/pkg && \
```
Comment (saghul) [inline web/Dockerfile:12]: Unnecessary change, please undo

File: web/rootfs/defaults/meet.conf [hunk 4]
```diff
@@ -22,6 +22,10 @@ add_header X-XSS-Protection "1; mode=block";
 add_header X-Jitsi-Shard {{ .Env.DEPLOYMENTINFO_SHARD }};
 {{ end }}
 
+{{ if .Env.DISABLE_FLOC | default "1" | toBool }}
```
Comment (saghul) [inline web/rootfs/defaults/meet.conf:25]: This is very counter intuitive. Since it's opt-in, let's call it ENABLE_FLOC instead, and default to 0.

**Review by saghul — CHANGES_REQUESTED** [review]
Left some feedback, thanks for doing this!

File: examples/traefik-v2/docker-compose.yml (hunk 1 above)
Comment (elrido) [reply examples/traefik-v2/docker-compose.yml]: Done in 9c5a4ac

File: examples/traefik/docker-compose.yml (hunk 2 above)
Comment (elrido) [reply examples/traefik/docker-compose.yml]: Done in 9c5a4ac

File: web/Dockerfile (hunk 3 above)
Comment (elrido) [reply web/Dockerfile]: Done in 9c5a4ac

File: web/rootfs/defaults/meet.conf (hunk 4 above)
Comment (elrido) [reply web/rootfs/defaults/meet.conf]: Thank you for the advice, inverted it in 9c5a4ac (incl. wording and commit msg)

File: web/rootfs/defaults/meet.conf [hunk 5]
```diff
@@ -22,6 +22,10 @@ add_header X-XSS-Protection "1; mode=block";
 add_header X-Jitsi-Shard {{ .Env.DEPLOYMENTINFO_SHARD }};
 {{ end }}
 
+{{ if .Env.ENABLE_FLOC | default "0" | toBool }}
+add_header Permissions-Policy "interest-cohort=()";
```
Comment (saghul) [inline web/rootfs/defaults/meet.conf:26]: AFAIS in https://plausible.io/blog/google-floc the header must be set to opt-out, so the if needs to be reversed, and set the header when the enabled flag has NOT been set.

**Review by saghul — CHANGES_REQUESTED** [review]
The if needs reversing

File: web/rootfs/defaults/meet.conf (hunk 5 above)
Comment (elrido) [reply web/rootfs/defaults/meet.conf:26]: Thank you for spotting this, I missed that while inverting the logic. I hope I got the frep/golang syntax correct in efc3f30

**Review by saghul — APPROVED** [review]

**Comment by saghul** [comment]
Cheers!
